core/src/parser.rs

```diff
@@ -618,54 +636,125 @@ fn is_redacted(attrs: &[syn::Attribute]) -> bool {
     })
 }
 
-/// Check if we have a `target_os` cfg that dooes not match command line
-/// argument `--target-os`.
+/// Check if this attribute rejects our optional `target_os` arguments.
+///
+/// Examples:
+/// - `#[cfg(not(target_os = "windows"))]`
+/// - `#[cfg(not(any(target_os = "windows", target_os = "macos", feature = "my-feature")))]`
+/// - `#[cfg(not(all(target_os = "windows", feature = "my-feature")))]`
 #[inline]
-pub(crate) fn target_os_skip(attr: &Attribute, target_os: &str) -> bool {
-    get_meta_items(attr, "cfg")
-        .find_map(|meta| match &meta {
-            // a single #[cfg(target_os = "target")]
-            Meta::NameValue(MetaNameValue {
-                path,
-                value:
-                    Expr::Lit(ExprLit {
-                        lit: Lit::Str(v), ..
-                    }),
-                ..
-            }) if path.is_ident("target_os") => Some(v.value()),
-            // combined with any or all
-            // Ex: #[cfg(any(target_os = "target", feature = "test"))]
-            Meta::List(meta_list)
-                if meta_list.path.is_ident("any") || meta_list.path.is_ident("all") =>
-            {
-                target_os_from_meta_list(meta_list)
+pub(crate) fn target_os_reject(attr: &Attribute, target_os: &str) -> bool {
+    if log_enabled!(log::Level::Debug) {
+        debug!(
+            "\tchecking attribute {} for {target_os} reject",
+            attr.into_token_stream()
+        );
+    }
+
+    let reject = get_meta_items(attr, "cfg").any(|meta| match &meta {
```

Comment:
Does `get_meta_items` check recursively through nested lists? Specifically I'm wondering if this would work for the case of:

```
#[cfg(all(feature = "my-feature", not(target_os = "windows")))]
```

To reject Windows as a target OS.

Reply:
I initially was going to avoid that, however I reworked it to handle recursive definitions now.